src/plugins/data/server/search/opensearch_search/decide_client.ts

```diff
@@ -9,14 +9,14 @@ import { IOpenSearchSearchRequest } from '..';
 export const decideClient = async (
   context: RequestHandlerContext,
   request: IOpenSearchSearchRequest,
+  withDataSourceEnabled: boolean = false,
   withLongNumeralsSupport: boolean = false
 ): Promise<OpenSearchClient> => {
-  // if data source feature is disabled, return default opensearch client of current user
-  const client =
-    request.dataSourceId && context.dataSource
-      ? await context.dataSource.opensearch.getClient(request.dataSourceId)
-      : withLongNumeralsSupport
-      ? context.core.opensearch.client.asCurrentUserWithLongNumeralsSupport
-      : context.core.opensearch.client.asCurrentUser;
-  return client;
+  const defaultOpenSearchClient = withLongNumeralsSupport
+    ? context.core.opensearch.client.asCurrentUserWithLongNumeralsSupport
+    : context.core.opensearch.client.asCurrentUser;
+
+  return withDataSourceEnabled && request.dataSourceId && context.dataSource
```

Comment:
can we assume when data source is enabled, then context.dataSource is never undefined or vice versa?

Reply:
Thanks for the comments~

Yes, we can assume when data source is enabled, then context.dataSource is never undefined or vice versa because:

1. "dataSource" will always been registered as "registerRouteHandlerContext" in [dataSource plugin setup](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/790c076fd4c8f38ade5482cede6d92d10792be54/src/plugins/data_source/server/plugin.ts#L115) which means once dataSource plugin's setup function get called, ```context.dataSource``` will never be undefined.
2. If [dataSource.enabled ](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/config/opensearch_dashboards.yml#L239) turns on,  [dataSource plugin's setup function](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/790c076fd4c8f38ade5482cede6d92d10792be54/src/plugins/data_source/server/plugin.ts#L52) will always get called.

In that case, ```withDataSourceEnabled``` is not needed in this decide client function. I will remove it.

Thanks